src/sagemaker/lineage/artifact.py

```diff
@@ -159,6 +159,57 @@ def downstream_trials(self, sagemaker_session=None) -> list:
         )
         trial_component_arns: list = list(map(lambda x: x.destination_arn, outgoing_associations))
 
+        return self._get_trial_from_trial_component(trial_component_arns)
+
+    def downstream_trials_v2(
+        self, direction: LineageQueryDirectionEnum = LineageQueryDirectionEnum.DESCENDANTS
```

Comment:
Same comment as the other PR. For methods that explicitly callout "upstream" or "downstream", don't allow direction to be set.

Can we have a `_trials()` method that has all of the shared code between `downstream_trials_v2` and `upstream_trials`?